signer/tests/integration/transaction_signer.rs

```diff
@@ -508,3 +514,107 @@ async fn signing_set_validation_check_for_stacks_transactions() {
+        .with_mocked_stacks_client()
+        .build();
+
+    // Build the test context with mocked clients
+    ctx.with_bitcoin_client(|client| {
+        client
+            .expect_estimate_fee_rate()
+            .times(2)
+            .returning(|| Box::pin(async { Ok(1.3) }));
+    })
+    .await;
+
+    let (rpc, faucet) = sbtc::testing::regtest::initialize_blockchain();
+
+    // Create a test setup with a confirmed deposit transaction
+    let mut setup = TestSweepSetup::new_setup(rpc, faucet, 10000, &mut rng);
+
+    // Backfill the blockchain data into the database
+    let chain_tip: BitcoinBlockHash = setup.sweep_block_hash.into();
+    backfill_bitcoin_blocks(&db, rpc, &chain_tip).await;
+    let bitcoin_block = db.get_bitcoin_block(&chain_tip).await.unwrap();
+
+    let public_aggregate_key = setup.aggregated_signer.keypair.public_key().into();
+
+    // Fill the signer's UTXO in the database
+    fill_signers_utxo(&db, bitcoin_block.unwrap(), &public_aggregate_key, &mut rng).await;
+
+    // Store the necessary data for passing validation
+    setup.store_happy_path_data(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Initialize the transaction signer event loop
+    let network = InMemoryNetwork::new();
+    let mut tx_signer = TxSignerEventLoop {
+        network: network.connect(),
+        context: ctx.clone(),
+        context_window: 10000,
+        wsts_state_machines: HashMap::new(),
+        signer_private_key: PrivateKey::new(&mut rng),
+        threshold: 2,
+        rng: rand::rngs::StdRng::seed_from_u64(51),
+    };
+
+    let sbtc_requests: TxRequestIds = TxRequestIds {
+        deposits: vec![setup.deposit_request.outpoint.into()],
+        withdrawals: vec![],
+    };
+
+    let sbtc_context = SbtcRequestsContextMessage {
+        requests: vec![sbtc_requests],
+        fee_rate: 10.0,
+        last_fees: None,
+    };
+    let result = tx_signer
+        .handle_sbtc_requests_context_message_tester(&sbtc_context, &chain_tip)
+        .await;
+
+    assert!(result.is_ok());
+
+    let sbtc_state = tx_signer
+        .get_btc_state(&chain_tip, &public_aggregate_key)
+        .await
+        .unwrap();
+
+    // Create an unsigned transaction with the deposit request
+    // to obtain the sighashes and corresponding txid that should
+    // be stored in the database
+    let unsigned_tx = UnsignedTransaction::new(
+        Requests::new(vec![RequestRef::Deposit(&setup.deposit_request)]),
+        &sbtc_state,
+    )
+    .unwrap();
+
+    let digests = unsigned_tx.construct_digests().unwrap();
+    let signer_digest = digests.signer_sighash();
+    let deposit_digest = digests.deposit_sighashes();
+    assert_eq!(deposit_digest.len(), 1);
+    let deposit_digest = deposit_digest[0];
+
+    // Check that the sighashes are stored in the database
+    assert!(db
+        .will_sign_bitcoin_tx_sighash(&signer_digest.sighash.into())
+        .await
+        .is_ok());
```

Comment:
I don't think that this checks that the sighashes are stored in the database, just that the query returned successfully.